mmv1/products/discoveryengine/CmekConfig.yaml

```
@@ -0,0 +1,143 @@
+description: |
+  CmekConfig represents configurations used to enable CMEK data encryption with
+  Cloud KMS keys.
+references:
+  guides:
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/CmekConfig'
+base_url: 'projects/{{project}}/locations/{{location}}/cmekConfigs'
+self_link: 'projects/{{project}}/locations/{{location}}/cmekConfigs/{{cmek_config_id}}'
+# Update API is also used as create API.
+create_url: 'projects/{{project}}/locations/{{location}}/cmekConfigs/{{cmek_config_id}}?&setDefault={{set_default}}'
+create_verb: 'PATCH'
+update_url: 'projects/{{project}}/locations/{{location}}/cmekConfigs/{{cmek_config_id}}'
+update_verb: 'PATCH'
+delete_url: 'projects/{{project}}/locations/{{location}}/cmekConfigs/{{cmek_config_id}}'
+import_format:
+  - 'projects/{{project}}/locations/{{location}}/cmekConfigs/{{cmek_config_id}}'
+timeouts:
+  insert_minutes: 60
+  update_minutes: 60
+  delete_minutes: 60
+autogen_async: false
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+    timeouts:
+      insert_minutes: 60
+      update_minutes: 60
+      delete_minutes: 60
+  result:
+    resource_inside_response: true
+examples:
+  - name: 'discoveryengine_cmekconfig_default'
+    primary_resource_id: 'default'
+    primary_resource_name: 'fmt.Sprintf("tf_test_cmek_config%s", context["random_suffix"])'
+    vars:
+      cmek_config_id: 'cmek-config-id'
+      kms_key_name: 'kms-key-name'
+      kms_key_ring_name: 'kms-key-ring-name'
+    ignore_read_extra:
+      - 'project'
+parameters:
+  - name: 'location'
+    type: String
+    description: |
+      The geographic location where the CMEK config should reside. The value can
+      only be one of "us" and "eu".
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'cmekConfigId'
+    type: String
+    description: |
+      The unique id of the cmek config.
+    url_param_only: true
+    immutable: true
+  - name: 'setDefault'
+    type: Boolean
+    description: |
+      Set the following CmekConfig as the default to be used for child resources
+      if one is not specified.
+    url_param_only: true
+properties:
+  - name: 'name'
+    type: String
+    description: |
+      The unique full resource name of the cmek config. Values are of the format
+      `projects/{project}/locations/{location}/cmekConfigs/{cmek_config_id}`.
+      This field must be a UTF-8 encoded string with a length limit of 1024
+      characters.
+    output: true
+  - name: 'kmsKey'
+    type: String
+    description: |
+      KMS key resource name which will be used to encrypt resources
+      `projects/{project}/locations/{location}/keyRings/{keyRing}/cryptoKeys/{keyId}`.
+  - name: 'kmsKeyVersion'
+    type: String
+    description: |
+      KMS key version resource name which will be used to encrypt resources
+      `<kms_key>/cryptoKeyVersions/{keyVersion}`.
+  - name: 'state'
+    type: Enum
```

```suggestion
    type: String
```
Should be just String in Terraform given it's output-only. In terraform, enum is just implemented as string with validation for configurable values. For output-only fields, it's just string. We can also remove the enum_values list for this resource.